sentry-core/src/main/java/io/sentry/core/transport/ICurrentDateProvider.java

```diff
@@ -0,0 +1,22 @@
+package io.sentry.core.transport;
+
+import java.util.Date;
+
+/** Date Provider to make the Transport unit testable */
+interface ICurrentDateProvider {
+
+  /**
+   * Returns the current Date (local)
+   *
+   * @return the Date
+   */
+  Date getCurrentDate();
+
+  /**
+   * Returns a new Date based on the given timeMillis
+   *
+   * @param timeMillis the Time in millis
+   * @return the given Date
+   */
+  Date getDate(long timeMillis);
```

Comment:
I think we only need this because we add milliseconds to the current time, like `System.currentTimeMillis() + 1000`. What do you say we remove this method and add `currentTimeMillis` instead? Then we can do `new Date(currentDateProvider.currentTimeMillis() = 1000)`.

Reply:
`+` do you mean? sounds a good idea
what's your LOGAF scale here?